src/core/util.js

```diff
@@ -283,4 +285,221 @@ var parseTimeSeriesForC3 = function(graph_data) {
+            "units": data[model]['units']
+        };
+        flatData.push(modelInfo); 
+    }
+    return flatData;
+}
+
+/*
+    Helper function for exportTableDataToWorksheet, creates summary rows that appear at the top of the exported worksheet
+*/
+var createWorksheetSummaryCells = function(summary_data, time_of_year) {
+    // store worksheet cell contents to be later encoded as per output format
+    var cells = [];
+    var cell;
+    // write summary rows at top of worksheet
+    var summary_header = ["Model", "Emissions Scenario", "Time of Year", "Variable ID", "Variable Name"];
+    var summary_keys = ["model_id", "experiment", "time_of_year", "variable_id", "variable_name"];
+
+    var num_summary_rows = 3;
+    var num_summary_cols = summary_keys.length
+    for(var R = 0; R < num_summary_rows; ++R) {
+        for(var C = 0; C < num_summary_cols; ++C) {
+            if(R == 0) cell = {v: summary_header[C]};
+            else if(R == 1) {
+                if(summary_keys[C] == 'time_of_year') {
+                  cell = {v: time_of_year};      
+                }
+                else cell = {v: summary_data[summary_keys[C]]};
+            }
+            else cell = {v: ""};
+            cell.t = 's';
+            cells.push(cell);
+        }
+    }
+    return { 'num_rows': num_summary_rows, 'num_cols': num_summary_cols, 'cells': cells };
+}
+
+/*
+    Helper function for exportTableDataToWorksheet, creates data column headers and data entries for exported worksheet
+*/
+var fillWorksheetDataCells = function(data) {
+    // store worksheet cell contents to be later encoded as per output format
+    var cells = [];
+    var cell;
+    // populate worksheet with table data
+    var column_labels = ["Model Period", "Run", "Min", "Max", "W.Mean", "Median", "W.Std.Dev", "Units" ];
+    var data_keys = ["model_period", "run", "min", "max", "w_mean", "median", "w_stdev", "units" ];
+    var num_data_rows = Object.keys(data).length + 1;
+    var num_data_cols = data_keys.length
+
+    for(var R = 0; R < num_data_rows; ++R) {
+        for(var C = 0; C < num_data_cols; ++C) {
+            // create header row
+            if(R == 0) cell = {v: column_labels[C]};
+            // create cell object: .v is the actual data
+            else cell = {v: data[R-1][data_keys[C]]};
+            if(cell.v === null) continue;
+            // determine the cell type 
+            if(typeof cell.v === 'number') cell.t = 'n';
+            else cell.t = 's';
+            cells.push(cell);
+        }
+    }
+    return { 'num_rows': num_data_rows, 'num_cols': num_data_cols, 'cells': cells };
+}
+
+/*
+    Helper function for exportTableDataToWorksheet, combines summary rows, data column headers, and data into one worksheet
+*/
+var assembleWorksheet = function (summary_cells, data_cells) {
+    var R = 0;
+    var C = 0;
+    var cell;
+    var cell_ref;
+    var ws = {};
+
+    for(R = 0; R < summary_cells.num_rows; ++R) {
```

Comment:
Because the number of columns in the summary rows is different than the number of columns in the data rows, and the XLSX.utils.encode_cell() function needs the row & column indices of each valid cell.